### PR TITLE
Add InstallLegacyCRDs interceptor

### PR DESCRIPTION
**Pull request by colunira**
**Description**

To ensure we don't install CRDs from cluster-essentials it's needed to add an interceptor, that will always set the `global.installCRDs` flag to false.

Changes proposed in this pull request:

- Add InstallLegacyCRDs interceptor

**Related issue(s)**

Blocked by https://github.com/kyma-project/kyma/pull/10763, https://github.com/kyma-project/kyma/pull/11031

### CODE REVIEW

File: parallel-install/pkg/deployment/core.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (i *core) getConfig() (overrides.Provider, *engine.Engine, *engine.Engine,
 	overridesProvider, err := overrides.New(i.kubeClient, i.overrides.Map(), i.cfg.Log)
 
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("Failed to create overrides provider: exiting")
+		return nil, nil, nil, fmt.Errorf("Failed to create overrides provider: exiting: %s ", err)
```
Comment (clebs) [inline parallel-install/pkg/deployment/core.go:73]:
```suggestion
		return nil, nil, nil, errors.Wrap(err, "Failed to create overrides provider: exiting")
```

File: parallel-install/pkg/preinstaller/resourceparser.go [hunk 2]
```diff
@@ -87,10 +88,16 @@ func initializeDefaultDecoder() (runtime.Decoder, error) {
...
+
 	err = apiextv1.AddToScheme(sch)
 	if err != nil {
 		return nil, err
 	}
 
+	err = apiextv1beta1.AddToScheme(sch)
+	if err != nil {
+		return nil, err
+	}
```
Comment (clebs) [inline parallel-install/pkg/preinstaller/resourceparser.go:100]:
```suggestion
	if err := apiextv1beta1.AddToScheme(sch); err != nil {
		return nil, err
	}
```

**Review by strekm — APPROVED** [review]
/meow

**Review by tobiscr — APPROVED** [review]

**Review by clebs — APPROVED** [review]

**Pull request merged**